flink-libraries/flink-table/src/main/scala/org/apache/flink/api/java/table/JavaBatchTranslator.scala

```diff
@@ -41,21 +44,13 @@ class JavaBatchTranslator extends PlanTranslator {
 
     // create table representation from DataSet
     val dataSetTable = new DataSetTable[A](
-    repr.asInstanceOf[JavaDataSet[A]],
-    fieldNames
+      repr.asInstanceOf[JavaDataSet[A]],
+      fieldNames
     )
-
-    // register table in Cascading schema
-    val schema = Frameworks.createRootSchema(true)
     val tableName = repr.hashCode().toString
-    schema.add(tableName, dataSetTable)
 
-    // initialize RelBuilder
-    val frameworkConfig = Frameworks
-      .newConfigBuilder
-      .defaultSchema(schema)
-      .build
-    val relBuilder = RelBuilder.create(frameworkConfig)
+    TranslationContext.addDataSet(tableName, dataSetTable)
```

Comment:
Do we really want to have this static? What happens if we use multipleTableEnvironments in our program? They shouldn't influence each other. Is the hash code really unique? In the old Table API we had a AtomicCounter that guaranted uniqueness.